src/arviz_plots/plots/ppcplot.py

```diff
@@ -0,0 +1,479 @@
+          * "scatter" -> passed to :func: `~arviz_plots.visuals.scatter_x`
+
+        * "title" -> passed to :func:`~arviz_plots.visuals.labelled_title`
+        * "remove_axis" -> not passed anywhere, can only be ``False`` to skip calling this function
+
+    stats_kwargs : mapping, optional
+        Valid keys are:
+
+        * density -> passed to kde, cumulative, ...
+
+    pc_kwargs : mapping
+        Passed to :class:`arviz_plots.PlotCollection.wrap`
+
+    Returns
+    -------
+    PlotCollection
+
+    Examples
+    --------
+    WIP
+    """
+    if sample_dims is None:
+        sample_dims = rcParams["data.sample_dims"]
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+    if kind is None:
+        kind = rcParams["plot.density_kind"]
+    if plot_kwargs is None:
+        plot_kwargs = {}
+    if pc_kwargs is None:
+        pc_kwargs = {}
+    else:
+        pc_kwargs = pc_kwargs.copy()
+
+    if stats_kwargs is None:
+        stats_kwargs = {}
+    if facet_dims is None:
+        facet_dims = []
+
+    # check for duplication of facet_dims in top level arg input and pc_kwargs
+    if "cols" in pc_kwargs and len(facet_dims) > 0:
+        duplicated_dims = set(facet_dims).intersection(pc_kwargs["cols"])
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                message=f"""Facet dimensions have been defined twice.
+                Both in the the top level function arguments and in `pc_kwargs`. 
+                The `cols` key in `pc_kwargs` will take precedence.
+                                        
+                facet_dims = {facet_dims}
+                pc_kwargs["cols"] = {pc_kwargs["cols"]}
+                Duplicated dimensions: {duplicated_dims}.""",
+            )
+        # setting facet_dims to pc_kwargs defined values since it is used
+        # later to calculate dims to reduce
+        facet_dims = list(set(pc_kwargs["col"]).difference({"__variable__"}))
+
+    if group not in ("posterior", "prior"):
+        raise TypeError("`group` argument must be either `posterior` or `prior`")
+
+    predictive_data_group = f"{group}_predictive"
+    if observed is None:
+        observed = group == "posterior"  # by default true if posterior, false if prior
+
+    # checking if plot_kwargs["observed"] or plot_kwargs["aggregate"] is not inconsistent with
+    # top level bool arguments `observed`` and `aggregate`
+    if "observed" in plot_kwargs:
+        if plot_kwargs["observed"] != observed:
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
+                    "ignore",
+                    message="""`plot_kwargs['observed']` inconsistency detected.
+                    It is not the same as with the top level `observed` argument. 
+                    `plot_kwargs['observed']` will take precedence.""",
+                )
+            observed = plot_kwargs["observed"]
```

Comment:
I'm a bit confused by this logic (Wouldn't this imply if both are true, the error is raised?) Did you mean 
```if plot_kwargs.get("observed", False) is not False and not observed:```? (implying `plot_kwargs["observed"]` is set to true but `observed` is false, so an error should be raised as they contradict)

Also what about the condition when `observed` is true but `plot_kwargs["observed"]` is false- should we display the error for this condition too due to contradiction, or just let the top-level argument `observed` take precedence? 

Because it might be the case that `observed` is not set at all by the user but defaults to true (which happens in the code before this check) because of the posterior predictive group being plotted, but the user has set `plot_kwargs["observed"]` to false to not have it be displayed- in this case should the top level argument still take preference even though it took a default value and wasn't explicitly defined in the function call?

Reply:
You are right, the code makes no sense, sorry. I started writing with an idea on mind on how to do it then changed my mind but not the text.

The two options should be considered:

```python
if (
    observed and plot_kwargs.get("observed", True) is False
    or not observed and plot_kwargs.get("observed", False) is not False
):
    # raise warning or error
```

Reply:
Whenever they are incompatible we should raise an error, thus, none will ever take precedence over the other because they either mean the same thing or an error is thrown

Reply:
Actually, depending on #66 we cpuld simplify it to " `plot_kwargs["observed"] ` can't be False, use observed top level kwarg for that"

Reply:
I've used the first incompatibility logic for now- could we keep a note of this and make modifications after the proposals in #66 are finalized?